src/services/database.ts

```diff
@@ -93,3 +105,51 @@ const CriteriaIndexMap: Record<Exclude<SearchCriteria, SearchCriteria.ALL>, Tabl
   vin: 'VinIndex',
   trailerId: 'TrailerIdIndex',
 };
+export const postTechRecord = async (request: any) => {
+  const systemNumber = await generateNewNumber(NumberTypes.SystemNumber);
+  logger.info(`system number : ${systemNumber}`);
+  if (request.techRecord_vehicleType !== 'trl' && !request.primaryVrm) {
+    request.primaryVrm = await generateNewNumber(NumberTypes.ZNumber);
+  }
+  if (request.techRecord_vehicleType === 'trl' && request.techRecord.trailerId) {
+    request.trailerId = await generateNewNumber(NumberTypes.TrailerId);
+  } else if (request.techRecord_euVehicleCategory === ('o1' || 'o2')) {
+    request.trailerId = await generateNewNumber(NumberTypes.TNumber);
+  }
+  const { vin } = request;
+  request.systemNumber = systemNumber;
+  request.createdTimestamp = new Date().toISOString();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  request.partialVin = vin.length < 6 ? vin : vin.substring(request.vin.length - 6);
+  logger.info('request');
+  logger.info(request);
+  logger.info(`table name: ${tableName}`);
+  const command = {
+    TableName: tableName,
+    ConditionExpression: '#vin <> :vin AND #systemNumber <> :systemNumber',
+    ExpressionAttributeNames: {
+      '#vin': 'vin',
+      '#systemNumber': 'systemNumber',
+    },
+    ExpressionAttributeValues: {
+      ':vin': { S: vin },
+      ':systemNumber': { S: systemNumber },
+    },
+    Item: request,
+  };
+  logger.info('we have got to try and catch');
+  try {
+    const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);
```

Comment:
The bug is in this line, new class needs a new keyword. Ideally tho I would like to not use this and use the non document db client. Should be able to marshall it and use that? Unsure if there is something I am missing here however